packages/angular/src/wallet/wallet.store.ts

```diff
@@ -0,0 +1,295 @@
+import { Inject, Injectable, Optional } from '@angular/core';
+import { ComponentStore } from '@ngrx/component-store';
+import { SendTransactionOptions, WalletNotConnectedError, WalletNotReadyError } from '@solana/wallet-adapter-base';
+import { WalletName } from '@solana/wallet-adapter-wallets';
+import { Connection, Transaction } from '@solana/web3.js';
+import { asyncScheduler, combineLatest, defer, from, Observable, of, throwError } from 'rxjs';
+import { catchError, concatMap, filter, first, map, observeOn, switchMap, tap, withLatestFrom } from 'rxjs/operators';
+
+import { fromAdapterEvent, isNotNull } from '../operators';
+import { WalletNotSelectedError } from './wallet.errors';
+import { WALLET_CONFIG } from './wallet.tokens';
+import { WalletConfig, WalletState } from './wallet.types';
+
+export const WALLET_DEFAULT_CONFIG: WalletConfig = {
+    wallets: [],
+    autoConnect: false,
+    localStorageKey: 'walletName',
+    onError: (error: unknown) => console.error(error),
+};
+
+@Injectable()
+export class WalletStore extends ComponentStore<WalletState> {
+    private readonly _autoConnect = this._config?.autoConnect || false;
+    private readonly _localStorageKey = this._config?.localStorageKey || 'walletName';
+    private logError = this._config?.onError || ((error: unknown) => console.error(error));
+    readonly wallets$ = this.select((state) => state.wallets);
+    readonly selectedWallet$ = this.select((state) => state.selectedWallet);
+    readonly connected$ = this.select((state) => state.connected);
+    readonly connecting$ = this.select((state) => state.connecting);
+    readonly disconnecting$ = this.select((state) => state.disconnecting);
+    readonly wallet$ = this.select((state) => state.wallet);
+    readonly adapter$ = this.select((state) => state.adapter);
+    readonly publicKey$ = this.select((state) => state.publicKey);
+    readonly ready$ = this.select((state) => state.ready);
+    readonly anchorWallet$ = this.select(
+        this.publicKey$.pipe(isNotNull),
+        this.adapter$.pipe(isNotNull),
+        (publicKey, adapter) => {
+            return publicKey && 'signTransaction' in adapter && 'signAllTransactions' in adapter
+                ? {
+                      publicKey,
+                      signTransaction: (transaction: Transaction) => this.signTransaction(transaction)?.toPromise(),
+                      signAllTransactions: (transactions: Transaction[]) =>
+                          this.signAllTransactions(transactions)?.toPromise(),
+                  }
+                : undefined;
+        }
+    );
+
+    constructor(
+        @Optional()
+        @Inject(WALLET_CONFIG)
+        private _config: WalletConfig
```

Comment:
Can this only be provided in the constructor, or is there a way to change the config values afterward?

Reply:
As it is, you can't change the values afterward. 

I can add a `setConfig` method to allow consumers to change it later, no biggie. The reasoning behind putting this in the constructor is because it's what you usually see in Angular maintained libraries, I'd keep it while also allowing consumers to change it on the fly.

Reply:
Yeah, it's probably fine to leave it as is. If we see a need for adding the ability to change it, we can always add later.